src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,923 @@
+		log.Println("response Headers:", resp.Header)
+		log.Println("response Body:", string(body))
+	}
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/monit.go#L813
+func createAnnotation() {
+
+	if annotation != "" {
+		if tags != "" {
+			dashboards := findDashboard()
+			timeRanges := parseTimes(true) // True because we are creating annotation and time ranges should be in future
+			if trange == "" {
+				log.Fatalf("Unable to create Annotation. Please provide Time Range!\n")
+			}
+			if configJSON.Verbose > 0 {
+				log.Println("timeRanges", timeRanges)
+			}
+			for _, r := range dashboards {
+				rec := make(map[string]interface{})
+				rec["dashboardId"] = r["id"]
+				rec["time"] = timeRanges[0]
+				rec["timeEnd"] = timeRanges[1]
+				rec["tags"] = parseTags(tags)
+				rec["text"] = annotation
+				data, err := json.Marshal(rec)
+				if err != nil {
+					log.Fatalf("Unable to marshal the data %+v, error %v\n", rec, err)
+				}
+				if configJSON.Verbose > 0 {
+					log.Printf("Add annotation: %+v", rec)
+				}
+				addAnnotation(data)
+			}
+
+			log.Printf("Annotation Created Successfully\n")
+
+		} else {
+			log.Fatalf("Can't Create annotation. Please provide tags!\n")
+		}
+	} else {
+		log.Fatalf("Can't Create annotation without any text!\n")
+	}
+}
+
+// helper function for updating an annotation which makes PUT request
+func updateAnnotationHelper(annotationID int, data []byte) error {
+
+	var headers [][]string
+	bearer := fmt.Sprintf("Bearer %s", configJSON.Token)
+	h := []string{"Authorization", bearer}
+	headers = append(headers, h)
+	h = []string{"Content-Type", "application/json"}
+	headers = append(headers, h)
+	h = []string{"Accept", "application/json"}
+	headers = append(headers, h)
+
+	apiurl := fmt.Sprintf("%s%s%d", configJSON.GrafanaBaseURL, configJSON.DeleteAnnotationAPI, annotationID)
+	if configJSON.Verbose > 0 {
+		log.Println(apiurl)
+	}
+	req, err := http.NewRequest("PUT", apiurl, bytes.NewBuffer(data))
+	if err != nil {
+		log.Printf("Unable to make request to %s, error: %s", apiurl, err)
+		return err
+	}
+	for _, v := range headers {
+		if len(v) == 2 {
+			req.Header.Add(v[0], v[1])
+		}
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("request: ", string(dump))
+		}
+	}
+	timeout := time.Duration(configJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Printf("Unable to get response from %s, error: %s", apiurl, err)
+		return err
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("response:", string(dump))
+		}
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("HTTP Respose error, code: %d", resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
could you please handle the error properly here and check the code that all errors will be logged

Reply:
Fixed !